server/lib/nl/detection/detector.py

```diff
@@ -142,7 +142,10 @@ def construct(entities: List[str], vars: List[str], child_type: str,
       if not any([child_type == x.value for x in types.ContainedInPlaceType]):
         counters.err('failed_detection_badChildEntityType', child_type)
         return None, f'Bad childEntityType value {child_type}!'
-      child_type = types.ContainedInPlaceType(child_type)
+      if child_type == places[0].place_type:
+        child_type = None
+      else:
```

Comment:
Why is this needed?

Reply:
when calling fulfill on the client code, we pass in the place type from the saved context (unless we don't want to do this?) & the passed in place type could end up being the same type as the place, which will then cause state.place_type to get set to None here: https://github.com/datacommonsorg/website/blob/f8b5fec8ecd06a758c3cfc8cfa8b2ada21ec034c/server/lib/nl/fulfillment/utils.py#L141 which then eventually, an error gets thrown when trying to access state.place_type

Reply:
What is the eventual error that gets thrown?  Because the setting of state.place_type to None is intentional, and that downstream error should be fixed.

If user has not asked for "place_type" then it should be empty, and we'll default to DEFAULT_TYPE.

If the user has said "countries in USA", then we set place_type to empty, and only show US-level data.

Reply:
reverted & updated the frontend code as we talked about offline